.github/workflows/docker-publish.yml

```
@@ -0,0 +1,35 @@
+name: Semantic Release
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  Release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Cache Node.js modules
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Semantic Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GA_TOKEN }}
```

You should use the [default token created for the workflow](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) and [adjust the permissions assigned to it](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) instead.
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```